test/test_common/logging.h

```diff
@@ -201,4 +201,31 @@ using ExpectedLogMessages = std::vector<StringPair>;
     ASSERT_EQ(0, logs.size()) << " Logs:\n   " << absl::StrJoin(logs, "   ");                      \
   } while (false)
 
+// Validates that when stmt is executed, the supplied substring is eventually logged.
+// This both waits infinitely (not ideal) and walks the full log list every time (not ideal)
```

Comment:
But other than that, this is ideal, right? :P In any case, nit: please add a "." to the end of the comment.